src/core.ts

```diff
@@ -282,13 +282,16 @@ function areFunctionsFullyImplemented(contract: ContractDefinition, deref: ASTDe
 }
 
 function getFunctionId(fn: FunctionDefinition, context: ContractDefinition, deref: ASTDereferencer): string {
-  const storageArgs = new Set<Argument>(getStorageArguments(fn, context, deref));
-  const nonStorageArgs = getFunctionArguments(fn, context, deref).filter(a => !storageArgs.has(a));
-  return fn.name + nonStorageArgs.map(a => a.type).join('');
+  const abiTypes = getFunctionArguments(fn, context, deref).map(a => a.abiType);
+  return fn.name + abiTypes.join(',');
 }
 
-function getFunctionNameQualified(fn: FunctionDefinition, context: ContractDefinition, deref: ASTDereferencer, onlyStorage: boolean = true): string {
-  return fn.name + (onlyStorage ? getStorageArguments(fn, context, deref) : getFunctionArguments(fn, context, deref))
+function getFunctionNameQualified(fn: FunctionDefinition, context: ContractDefinition, deref: ASTDereferencer, onlyConflicting: boolean): string {
+  let args = getFunctionArguments(fn, context, deref);
+  if (onlyConflicting) {
+    args = args.filter(a => a.type !== a.abiType || a.storageType !== undefined);
+  }
+  return fn.name + args
```

Comment:
```suggestion
  return fn.name + getFunctionArguments(fn, context, deref)
    .filter(a => !onlyConflicting || a.type !== a.abiType || a.storageType !== undefined)
```

Reply:
I find the current code easier to reason about.